### PR TITLE
Refactor: remove d3 selection multi

### PR DESCRIPTION
**Pull request by maartenbreddels**
`d3-selection-multi` monkey patches `d3-selection`, this does not work well
with federated modules, since another provider may provide a semver
compatible d3 that is unpacked. This causes issues with jupyterlab in combination with other widgets/extensions that use (and thus *can* provide) d3.

Related: #1245 #1222

It seems https://github.com/bqplot/bqplot/pull/1222/files#diff-e51a40ac250c9696142466f114f754161e7e5102c0cdb5354548b757deb272f6R89 does not work at all, I may have been lucky in thinking it did before, or this is a regression in some tooling, in any case, we should get rid of this monkey patching library.

Note: there are many more uses of `.styles` and `.attrs`, opening this soon to get feedback.

### CODE REVIEW

File: js/package.json [hunk 1]
```diff
@@ -85,8 +85,7 @@
       "@jupyter-widgets/base": {
         "bundled": false,
         "singleton": true
-      },
-      "d3": false
```
Comment (martinRenou) [inline js/package.json:89]: Do we need to remove this?

**Comment by JohanMabille** [comment]
I'm totally in favor of this approach. Besides, `d3-selection-multi` has been archived and is not maintained anymore.

File: js/package.json (hunk 1 above)
Comment (maartenbreddels) [reply js/package.json:89]: yes!

Comment (maartenbreddels) [reply js/package.json:89]: Hahah, I thought you reminder me to remove this, I misunderstood.
Yes, we can remove this because we can share d3 now.

**Comment by dhirschfeld** [comment]
> It seems #1222 (files) does not work at all

Does this mean https://github.com/bqplot/bqplot/releases/tag/0.12.20 is broken on jlab3?

**Comment by maartenbreddels** [comment]
It works when no other federated module provides d3, so by itself bqplot works. It might not work when others are installed.

**Comment by SylvainCorlay** [comment]
Thanks! Sorry for not merging earlier!

**Pull request merged**